src/Stryker.Core/Stryker.Core/ProjectComponents/TestProjects/TestFile.cs

```diff
@@ -1,13 +1,34 @@
 using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Stryker.Core.ProjectComponents.TestProjects
 {
     public sealed class TestFile : IEquatable<TestFile>
     {
         public string FilePath { get; init; }
         public string Source { get; init; }
-        public IEnumerable<TestCase> Tests { get; set; } = new List<TestCase>();
+        public IEnumerable<TestCase> Tests { get; private set; } = new List<TestCase>();
+
+        public void AddTest(Guid id, string name, string source, int lineNumber)
+        {
+            if (Tests.Any(test => test.Id == id))
+            {
+                return;
+            }
+
+            var tests = new List<TestCase>(Tests);
```

Comment:
Just wondering, why do you need all those temporary operations instead of just adjusting the code to add things directly to the existing property?

Reply:
No reason other than to try and avoid casting. But it's not really neccesary.